crates/nu-command/src/math/bit_not.rs

```diff
@@ -0,0 +1,80 @@
+use nu_protocol::ast::Call;
+use nu_protocol::engine::{Command, EngineState, Stack};
+use nu_protocol::{Category, Example, PipelineData, ShellError, Signature, Span, Value};
+
+#[derive(Clone)]
+pub struct SubCommand;
+
+impl Command for SubCommand {
+    fn name(&self) -> &str {
+        "math bit-not"
+    }
+
+    fn signature(&self) -> Signature {
+        Signature::build("math bit-not").category(Category::Math)
+    }
+
+    fn usage(&self) -> &str {
+        "performs logical negation on each bit"
+    }
+
+    fn search_terms(&self) -> Vec<&str> {
+        vec!["negation"]
+    }
+
+    fn run(
+        &self,
+        engine_state: &EngineState,
+        _stack: &mut Stack,
+        call: &Call,
+        input: PipelineData,
+    ) -> Result<nu_protocol::PipelineData, nu_protocol::ShellError> {
+        let head = call.head;
+        input.map(
+            move |value| operate(value, head),
+            engine_state.ctrlc.clone(),
+        )
+    }
+
+    fn examples(&self) -> Vec<Example> {
+        vec![Example {
+            description: "Apply the floor function to a list of numbers",
+            example: "[4 3 2] | math bit-not",
+            result: Some(Value::List {
+                vals: vec![
+                    Value::test_int(-5),
+                    Value::test_int(-4),
+                    Value::test_int(-3),
+                ],
+                span: Span::test_data(),
+            }),
+        }]
+    }
+}
+
+fn operate(value: Value, head: Span) -> Value {
+    match value {
+        Value::Int { val, span } => Value::Int { val: !val, span },
```

Comment:
This assumes the underlying representation should always be `i64` with two's complement arithmetic for negative values.

Reply:
Thanks, that's something I never thought about.

To handle this, I've add `--signed`, so user can make their input integer as a signed integer
And `--number-size` flag, to make user tell `nu` treat a unsigned number as `1-bytes/2-bytes/4-bytes/8-bytes number`

Reply:
Forgot my two's complement math. For signed integers the size shouldn't matter as a larger datatype would result in the same sign extension. Word size only matters for the unsigned case

Reply:
yeah, this is current implementation

Reply:
Can we maybe replace `--number-size` with `--num-bytes` else it could also be bits. Apart from that it seems ready to land and test for me

Reply:
Ok, renamed it